internal/elasticsearch/transform/transform.go

```diff
@@ -0,0 +1,843 @@
+									Type:         schema.TypeString,
+									Required:     true,
+									ValidateFunc: utils.StringIsElasticDuration,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"sync": {
+			Description: "Defines the properties transforms require to run continuously.",
+			Type:        schema.TypeList,
+			Optional:    true,
+			MaxItems:    1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"time": {
+						Description: "Specifies that the transform uses a time field to synchronize the source and destination indices. This is currently the only supported option.",
+						Type:        schema.TypeList,
+						Required:    true,
+						MaxItems:    1,
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								"field": {
+									Description: "The date field that is used to identify new documents in the source.",
+									Type:        schema.TypeString,
+									Required:    true,
+								},
+								"delay": {
+									Description:  "The time delay between the current time and the latest input data time. The default value is 60s.",
+									Type:         schema.TypeString,
+									Optional:     true,
+									Default:      "60s",
+									ValidateFunc: utils.StringIsElasticDuration,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"align_checkpoints": {
+			Description: "Specifies whether the transform checkpoint ranges should be optimized for performance.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+		},
+		"dates_as_epoch_millis": {
+			Description: "Defines if dates in the output should be written as ISO formatted string (default) or as millis since epoch.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+		},
+		"deduce_mappings": {
+			Description: "Specifies whether the transform should deduce the destination index mappings from the transform config.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+		},
+		"docs_per_second": {
+			Description:  "Specifies a limit on the number of input documents per second. Default (unset) value disables throttling.",
+			Type:         schema.TypeFloat,
+			Optional:     true,
+			Default:      -1,
+			ValidateFunc: validation.FloatAtLeast(0),
+		},
+		"max_page_search_size": {
+			Description:  "Defines the initial page size to use for the composite aggregation for each checkpoint. Default is 500.",
+			Type:         schema.TypeInt,
+			Optional:     true,
+			Default:      -1,
+			ValidateFunc: validation.IntBetween(10, 65536),
+		},
+		"num_failure_retries": {
+			Description:  "Defines the number of retries on a recoverable failure before the transform task is marked as failed. The default value is the cluster-level setting num_transform_failure_retries.",
+			Type:         schema.TypeInt,
+			Optional:     true,
+			Default:      -2,
```

Comment:
This isn't the real default. It sounds like we can just remove the default value here. 
```suggestion
```